src/System.Diagnostics.TraceSource/tests/DefaultTraceListenerClassTests.cs

```diff
@@ -45,6 +60,46 @@ public void Fail()
         }
 
         [Fact]
+        public void Fail_WithoutWriteLineOverride()
+        {
+            var listener = new TestDefaultTraceListener();
+            listener.ShouldOverrideWriteLine = false;
+            listener.Fail("FAIL");
+            listener.ShouldOverrideWriteLine = true;
+            Assert.False(listener.Output.Contains("FAIL"));
+        }
+
+        [Fact]
+        public void Fail_WithLogFile()
+        {
+            var listener = new TestDefaultTraceListener();
+            listener.LogFileName = "LogFile.txt";
+
+            string pathToLogFile = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), listener.LogFileName);
```

Comment:
It'd be better not to have the log file go to the execution directory, a) because it clutters up the dev's execution environment, and b) because I believe that won't work in certain environments (cc: @danmosemsft, who just hit this on some other tests).  I assume LogFileName can be set to an absolute path?  In that case I suggest having the test class derive from FileCleanupTestBase (you can see examples of this in other test classes), and then do:
```C#
string pathToLogFile = GetTestFilePath();
listener.LogFileName = pathToLogFile;
```
You then also don't need to delete it later, as that's handled by the base class.